test-integration/operator-tests/src/test/java/org/infinispan/operator/AdvancedSetupAIT.java

```diff
@@ -0,0 +1,194 @@
+package org.infinispan.operator;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.function.BooleanSupplier;
+
+import org.apache.http.entity.ContentType;
+import org.assertj.core.api.Assertions;
+import org.infinispan.Caches;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.client.hotrod.configuration.ClientIntelligence;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.exceptions.HotRodClientException;
+import org.infinispan.commons.configuration.XMLStringConfiguration;
+import org.infinispan.identities.Credentials;
+import org.infinispan.util.KeystoreGenerator;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import cz.xtf.builder.builders.SecretBuilder;
+import cz.xtf.client.Http;
+import cz.xtf.core.http.Https;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.core.waiting.SimpleWaiter;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.Secret;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Compared to MinimalSetupIT this set of tests are running
+ * against features that need to be configured.
+ *
+ * spec:
+ *   expose:
+ *     type: Route
+ *   container:
+ *     cpu: 1500m
+ *     memory: 1Gi
+ *   logging:
+ *     categories:
+ *       x.y.z: trace
+ *       t.u.v: debug
+ *   security:
+ *     endpointEncryption:
+ *        type: secret
+ *        certSecretName: encryption-secret
+ *     endpointSecretName: connect-secret
+ *   service:
+ *     type: datagrid
```

Comment:
Not related to code, just description side
```suggestion
 *     type: DataGrid
```